src/Microsoft.Diagnostics.EventFlow.Outputs.ApplicationInsights/ApplicationInsightsOutput.cs

```diff
@@ -145,21 +145,34 @@ private void Initialize(ApplicationInsightsOutputConfiguration aiOutputConfigura
             Debug.Assert(aiOutputConfiguration != null);
             Debug.Assert(this.healthReporter != null);
 
+            if (!aiOutputConfiguration.Validate(out string validationError))
+            {
+                this.healthReporter.ReportProblem($"{nameof(ApplicationInsightsOutput)}: invalid configuration. {validationError} No data will be sent to Application Insights", EventFlowContextIdentifiers.Output);
+                return;
+            }
+
+            TelemetryConfiguration telemetryConfiguration = null;
             if (string.IsNullOrWhiteSpace(aiOutputConfiguration.ConfigurationFilePath))
             {
-                this.telemetryClient = new TelemetryClient();                
+                telemetryConfiguration = TelemetryConfiguration.CreateDefault();              
             }
             else
             {
                 string configurationFileContent = File.ReadAllText(aiOutputConfiguration.ConfigurationFilePath);
-                TelemetryConfiguration telemetryConfiguration = TelemetryConfiguration.CreateFromConfiguration(configurationFileContent);
-                this.telemetryClient = new TelemetryClient(telemetryConfiguration);
+                telemetryConfiguration = TelemetryConfiguration.CreateFromConfiguration(configurationFileContent);
+            }
+
+            if (!string.IsNullOrWhiteSpace(aiOutputConfiguration.ConnectionString))
+            {
+                telemetryConfiguration.ConnectionString = aiOutputConfiguration.ConnectionString;
             }
 
             if (!string.IsNullOrWhiteSpace(aiOutputConfiguration.InstrumentationKey))
             {
-                this.telemetryClient.InstrumentationKey = aiOutputConfiguration.InstrumentationKey;
+                telemetryConfiguration.InstrumentationKey = aiOutputConfiguration.InstrumentationKey;
```

Comment:
The iKey will override the connection string. This is documented and checked by the unit tests: https://github.com/Azure/diagnostics-eventflow/pull/370/files#diff-1a4b80c1804a377adc457a7db59c63c44f8ad6d3492d0ce6ba2e51f184446ad6R43

Reply:
Does TelemetryConfiguration also override connection string if iKey is provided?
This is a bit surprising as connection string contains iKey and endpoint. But if user only specifies iKey, AI will use the default endpoint. Or maybe AI points to the endpoint in connection string but just override the iKey in the connection string?

Reply:
@yantang-msft if you set iKey on TelemetryClient, it will override the key that was passed via TelemetryConfiguration. https://github.com/microsoft/ApplicationInsights-dotnet/blob/develop/BASE/src/Microsoft.ApplicationInsights/TelemetryClient.cs#L469

The connection string is applied at TelemetryConfiguration level.